### PR TITLE
RichCommand: ShareItem

### PR DESCRIPTION
**Pull request by mafra99**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Relates #11481

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [x] Did you remove any strings from the en-us resource file?
   - [x] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
  - Share is hidden in zip files & recycle bin
  - Share is hidden if no item is selected
  - Share is hidden if invalid item is selected (e.g. a folder)
  - Share is visible if a single item is selected
  - Share is visible if multiple items are selected
  - Sharing items works with single files as well as with multiple files

### CODE REVIEW

File: src/Files.App/Actions/Content/Share/ShareItemAction.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+				}
+
+				dataRequest.Data.SetStorageItems(items, false);
+				dataRequestDeferral.Complete();
+
+				// TODO: Unhook the event somewhere
+			}
+
+			return Task.CompletedTask;
+		}
```
Comment (yaira2) [inline src/Files.App/Actions/Content/Share/ShareItemAction.cs:113]: A lot of this code should probably be in a helper 🤔

File: src/Files.App/Actions/Content/Share/ShareItemAction.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.Foundation;
+using Windows.Storage;
+
+namespace Files.App.Actions.Content.Share
+{
```
Comment (ferrariofilippo) [inline src/Files.App/Actions/Content/Share/ShareItemAction.cs:18]:
```suggestion

namespace Files.App.Actions.Content
{
```

File: src/Files.App/Actions/Content/Share/ShareItemAction.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	internal class ShareItemAction : ObservableObject, IAction
+	{
+		private readonly IContentPageContext context = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+		public string Label => "Share".GetLocalizedResource();
+
+		public string Description => "TODO: Need to be described.";
+
```
Comment (ferrariofilippo) [inline src/Files.App/Actions/Content/Share/ShareItemAction.cs:26]: Can you add a description so we don't have to come back to this in the future?

File: src/Files.App/Actions/Content/Share/ShareItemAction.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+		public RichGlyph Glyph { get; } = new RichGlyph(opacityStyle: "ColorIconShare");
+
+
```
Comment (ferrariofilippo) [inline src/Files.App/Actions/Content/Share/ShareItemAction.cs:29]:
```suggestion

```

File: src/Files.App/UserControls/InnerNavigationToolbar.xaml [hunk 5]
```diff
@@ -155,13 +155,13 @@
...
 					AccessKey="H"
-					Command="{x:Bind ViewModel.Share, Mode=OneWay}"
-					IsEnabled="{x:Bind converters:MultiBooleanConverter.AndConvert(ViewModel.CanShare, ViewModel.InstanceViewModel.CanShareInPage), Mode=OneWay, FallbackValue=False}"
-					Label="{helpers:ResourceString Name=BaseLayoutItemContextFlyoutShare/Text}"
+					Command="{x:Bind Commands.ShareItem, Mode=OneWay}"
+					IsEnabled="{x:Bind Commands.ShareItem.IsExecutable, Mode=OneWay, FallbackValue=False}"
+					Label="{x:Bind Commands.ShareItem.Label}"
 					LabelPosition="Collapsed"
-					ToolTipService.ToolTip="{helpers:ResourceString Name=BaseLayoutItemContextFlyoutShare/Text}">
+					ToolTipService.ToolTip="{x:Bind Commands.ShareItem.Label}">
```
Comment (ferrariofilippo) [inline src/Files.App/UserControls/InnerNavigationToolbar.xaml:162]: It might be better to use the description in the tooltip

**Review by ferrariofilippo — CHANGES_REQUESTED** [review]
As Yair mentioned, I think you should create an Helper

File: src/Files.App/Actions/Content/Share/ShareItemAction.cs (hunk 3 above)
Comment (yaira2) [reply src/Files.App/Actions/Content/Share/ShareItemAction.cs:26]: Descriptions will be added later, it should be left like this for now.

File: src/Files.App/UserControls/InnerNavigationToolbar.xaml (hunk 5 above)
Comment (yaira2) [reply src/Files.App/UserControls/InnerNavigationToolbar.xaml:162]: Label is correct because it'll show the shortcut as well.

**Review by ferrariofilippo — APPROVED** [review]
LGTM

**Pull request merged**